app/index.js

```diff
@@ -290,6 +293,14 @@ app.on('ready', () => installDevExtensions(isDev).then(() => {
       Menu.getApplicationMenu().popup(Math.ceil(x), Math.ceil(y));
     });
 
+    rpc.on('open context menu', () => {
+      const commands = getKeymaps().commands;
+      const contextMenuTemplate = editMenu(commands).submenu
+        .concat({type: 'separator'}, shellMenu(commands, createWindow).submenu);
+      const contextMenu = Menu.buildFromTemplate(contextMenuTemplate);
```

Comment:
Plugin can indirectly modify context menu because you reconstruct its template each time. Maybe enough and improvements about plugin decoration could be done in another PR.